src/prefect/runner/server.py

```diff
@@ -0,0 +1,74 @@
+import pendulum
+import uvicorn
+from prefect._vendor.fastapi import APIRouter, FastAPI, status
+from prefect._vendor.fastapi.responses import JSONResponse
+
+from prefect.settings import (
+    PREFECT_RUNNER_HEALTH_CHECK_DELAY,
+    PREFECT_RUNNER_SERVER_HOST,
+    PREFECT_RUNNER_SERVER_LOG_LEVEL,
+    PREFECT_RUNNER_SERVER_PORT,
+)
+
+
+def perform_health_check(runner, delay_threshold: int = None) -> JSONResponse:
+    if delay_threshold is None:
+        delay_threshold = PREFECT_RUNNER_HEALTH_CHECK_DELAY.value()
+
+    def _health_check():
+        now = pendulum.now("utc")
+        poll_delay = (now - runner.last_polled).total_seconds()
+
+        if poll_delay > delay_threshold:
+            return JSONResponse(
+                status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+                content={"message": "Runner is unresponsive at this time"},
+            )
+        return JSONResponse(status_code=status.HTTP_200_OK, content={"message": "OK"})
+
+    return _health_check
+
+
+def run_count(runner) -> int:
+    def _run_count():
+        run_count = len(runner._flow_run_process_map)
+        return run_count
+
+    return _run_count
+
+
+def shutdown(runner) -> int:
+    def _shutdown():
+        runner.should_stop = True
```

Comment:
Alternative approach could be to add a `@sync_compatible` decorator on `.stop`

Reply:
yea, stop is already sync compatible - the problem with that is that this route will run in a separate thread, causing the cancellation of async tasks to error out

Reply:
Ah I thought that `stop` was async. If you make `stop` async and add a `@sync_compatible` decorator, I think it will take care of the `no running event loop` error. I tested that change and it appeared to work. Making that update is a simple way of bringing in our concurrency tooling so that `stop` is run in a new thread with an event loop.

Reply:
@desertaxle nice, updated and it does appear to be working 👍